src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTagger.cs

```diff
@@ -19,56 +24,118 @@ internal sealed class InlineParameterNameHintsTagger : ITagger<IntraTextAdornmen
     {
         private readonly ITagAggregator<InlineParameterNameHintDataTag> _tagAggregator;
         private readonly ITextBuffer _buffer;
+        private readonly ITextView _textView;
+
+        /// <summary>
+        /// _cache stores the parameter hint tags in a global location 
+        /// </summary>
+        private readonly List<ITagSpan<IntraTextAdornmentTag>> _cache;
+
+        /// <summary>
+        /// Stores the snapshot of the state at which things change on the screen 
+        /// </summary>
+        private ITextSnapshot? _cacheSnapshot;
 
-        public event EventHandler<SnapshotSpanEventArgs> TagsChanged;
+        private readonly IClassificationFormatMap _formatMap;
+        private TextFormattingRunProperties? _format;
+        private readonly IClassificationType _hint;
+        private readonly ForegroundThreadAffinitizedObject _threadAffinitizedObject;
 
-        public InlineParameterNameHintsTagger(ITextBuffer buffer, ITagAggregator<InlineParameterNameHintDataTag> tagAggregator)
+        public event EventHandler<SnapshotSpanEventArgs>? TagsChanged;
+
+        public InlineParameterNameHintsTagger(InlineParameterNameHintsTaggerProvider taggerProvider, ITextView textView, ITextBuffer buffer, ITagAggregator<InlineParameterNameHintDataTag> tagAggregator)
         {
+            _cache = new List<ITagSpan<IntraTextAdornmentTag>>();
+            _threadAffinitizedObject = new ForegroundThreadAffinitizedObject(taggerProvider.ThreadingContext);
+            _textView = textView;
             _buffer = buffer;
             _tagAggregator = tagAggregator;
+            _formatMap = taggerProvider.ClassificationFormatMapService.GetClassificationFormatMap(textView);
+            _hint = taggerProvider.ClassificationTypeRegistryService.GetClassificationType(InlineParameterNameHintsTag.TagId);
+            _formatMap.ClassificationFormatMappingChanged += this.OnClassificationFormatMappingChanged;
             _tagAggregator.TagsChanged += OnTagAggregatorTagsChanged;
         }
 
+        private void OnClassificationFormatMappingChanged(object sender, EventArgs e)
+        {
+            _threadAffinitizedObject.AssertIsForeground();
+            if (_format != null)
+            {
+                _format = null;
+                _cache.Clear();
```

Comment:
Does this need to raise tags changed? Presumably the editor is going to be responding to the format change and fetching new tags, but if something happens out of order that might be bad.